### ISSUE TITLE
addons are getting executed when control plane is powered down and upgrade fails

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#25648]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

- Deploy a cluster - 1 etcd, 3 control plane and 2 worker nodes.
- Power down one control plane node. 
- Upgrade k8s version in the cluster from 1.15 to 1.16
- The cluster goes to Upgrading state. 
- The cluster is stuck in Updating state with error: `Provisioning incomplete, host(s) [<inactive-control-plane>] skipped because they could not be contacted; Provisioning incomplete, host(s) [<inactive-control-plane>] skipped because they could not be contacted`
- But the add ons jobs are executed before this failure is seen

**logs similar to these:**
```
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: Initiating Kubernetes cluster
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [certificates] GenerateServingCertificate is disabled, checking if there are unused kubelet certificates
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [certificates] Generating admin certificates and kubeconfig
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [certificates] Checking and deleting unused kube-etcd certificates
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: Successfully Deployed state file at [management-state/rke/rke-250920167/cluster.rkestate]
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: Building Kubernetes cluster
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [dialer] Setup tunnel for host [167.172.249.119]
2020/02/28 00:53:28 [ERROR] cluster [c-p8w2x] provisioning: Failed to set up SSH tunneling for host [167.172.249.119]: Can't establish dialer connection: can not build dialer to [c-p8w2x:m-jp2pf]
2020/02/28 00:53:28 [INFO] Checking if container [cert-deployer] is running on host [167.172.26.9], try #1



2020/02/28 00:53:28 [INFO] Checking if container [cert-deployer] is running on host [138.197.2.221], try #1



2020/02/28 00:53:28 [INFO] Checking if container [cert-deployer] is running on host [167.172.28.157], try #1



2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [dialer] Setup tunnel for host [167.172.28.157]
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [dialer] Setup tunnel for host [167.172.26.9]
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [dialer] Setup tunnel for host [138.197.2.221]
2020/02/28 00:53:28 [ERROR] cluster [c-p8w2x] provisioning: Removing host [167.172.249.119] from node lists
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [network] No hosts added existing cluster, skipping port check
2020/02/28 00:53:28 [INFO] cluster [c-p8w2x] provisioning: [certificates] Deploying kubernetes certificates to Cluster nodes
2020/02/28 00:53:28 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.28.157]
2020/02/28 00:53:28 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.26.9]
2020/02/28 00:53:28 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [138.197.2.221]
2020/02/28 00:53:28 [INFO] Starting container [cert-deployer] on host [167.172.28.157], try #1



2020/02/28 00:53:28 [INFO] Starting container [cert-deployer] on host [167.172.26.9], try #1



2020/02/28 00:53:28 [INFO] Starting container [cert-deployer] on host [138.197.2.221], try #1



2020/02/28 00:53:29 [INFO] Checking if container [cert-deployer] is running on host [167.172.26.9], try #1



2020/02/28 00:53:29 [INFO] Checking if container [cert-deployer] is running on host [167.172.28.157], try #1



2020/02/28 00:53:29 [INFO] Checking if container [cert-deployer] is running on host [138.197.2.221], try #1



2020/02/28 00:53:34 [INFO] Checking if container [cert-deployer] is running on host [167.172.26.9], try #1



2020/02/28 00:53:34 [INFO] Checking if container [cert-deployer] is running on host [167.172.28.157], try #1



2020/02/28 00:53:34 [INFO] Removing container [cert-deployer] on host [167.172.26.9], try #1



2020/02/28 00:53:34 [INFO] Checking if container [cert-deployer] is running on host [138.197.2.221], try #1



2020/02/28 00:53:34 [INFO] Removing container [cert-deployer] on host [167.172.28.157], try #1



2020/02/28 00:53:34 [INFO] Removing container [cert-deployer] on host [138.197.2.221], try #1



2020/02/28 00:53:34 [INFO] cluster [c-p8w2x] provisioning: [reconcile] Rebuilding and updating local kube config
I0228 00:53:34.618766       6 http.go:116] HTTP2 has been explicitly disabled
2020/02/28 00:53:34 [INFO] cluster [c-p8w2x] provisioning: Successfully Deployed local admin kubeconfig at [management-state/rke/rke-250920167/kube_config_cluster.yml]
2020/02/28 00:53:34 [INFO] cluster [c-p8w2x] provisioning: [reconcile] host [138.197.2.221] is active master on the cluster
2020/02/28 00:53:34 [INFO] cluster [c-p8w2x] provisioning: [certificates] Successfully deployed kubernetes certificates to Cluster nodes
2020/02/28 00:53:34 [INFO] cluster [c-p8w2x] provisioning: [file-deploy] Deploying file [/etc/kubernetes/kube-api-authn-webhook.yaml] to node [167.172.26.9]
2020/02/28 00:53:35 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.26.9]
2020/02/28 00:53:35 [INFO] Starting container [file-deployer] on host [167.172.26.9], try #1



2020/02/28 00:53:35 [INFO] cluster [c-p8w2x] provisioning: Successfully started [file-deployer] container on host [167.172.26.9]
2020/02/28 00:53:35 [INFO] cluster [c-p8w2x] provisioning: Waiting for [file-deployer] container to exit on host [167.172.26.9]
2020/02/28 00:53:35 [INFO] cluster [c-p8w2x] provisioning: Waiting for [file-deployer] container to exit on host [167.172.26.9]
2020/02/28 00:53:36 [INFO] Removing container [file-deployer] on host [167.172.26.9], try #1



2020/02/28 00:53:36 [INFO] cluster [c-p8w2x] provisioning: [remove/file-deployer] Successfully removed container on host [167.172.26.9]
2020/02/28 00:53:36 [INFO] cluster [c-p8w2x] provisioning: [file-deploy] Deploying file [/etc/kubernetes/kube-api-authn-webhook.yaml] to node [138.197.2.221]
2020/02/28 00:53:36 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [138.197.2.221]
2020/02/28 00:53:36 [INFO] Starting container [file-deployer] on host [138.197.2.221], try #1



2020/02/28 00:53:37 [INFO] cluster [c-p8w2x] provisioning: Successfully started [file-deployer] container on host [138.197.2.221]
2020/02/28 00:53:37 [INFO] cluster [c-p8w2x] provisioning: Waiting for [file-deployer] container to exit on host [138.197.2.221]
2020/02/28 00:53:37 [INFO] cluster [c-p8w2x] provisioning: Waiting for [file-deployer] container to exit on host [138.197.2.221]
2020/02/28 00:53:37 [INFO] Removing container [file-deployer] on host [138.197.2.221], try #1



2020/02/28 00:53:37 [INFO] cluster [c-p8w2x] provisioning: [remove/file-deployer] Successfully removed container on host [138.197.2.221]
2020/02/28 00:53:37 [INFO] cluster [c-p8w2x] provisioning: [file-deploy] Deploying file [/etc/kubernetes/kube-api-authn-webhook.yaml] to node [167.172.28.157]
2020/02/28 00:53:37 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.28.157]
2020/02/28 00:53:38 [INFO] Starting container [file-deployer] on host [167.172.28.157], try #1



2020/02/28 00:53:38 [INFO] cluster [c-p8w2x] provisioning: Successfully started [file-deployer] container on host [167.172.28.157]
2020/02/28 00:53:38 [INFO] cluster [c-p8w2x] provisioning: Waiting for [file-deployer] container to exit on host [167.172.28.157]
2020/02/28 00:53:38 [INFO] cluster [c-p8w2x] provisioning: Waiting for [file-deployer] container to exit on host [167.172.28.157]
2020/02/28 00:53:38 [INFO] Removing container [file-deployer] on host [167.172.28.157], try #1



2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [remove/file-deployer] Successfully removed container on host [167.172.28.157]
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [/etc/kubernetes/kube-api-authn-webhook.yaml] Successfully deployed authentication webhook config Cluster nodes
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [reconcile] Reconciling cluster state
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [reconcile] Check etcd hosts to be deleted
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [reconcile] Check etcd hosts to be added
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [reconcile] Rebuilding and updating local kube config
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: Successfully Deployed local admin kubeconfig at [management-state/rke/rke-250920167/kube_config_cluster.yml]
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [reconcile] host [138.197.2.221] is active master on the cluster
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [reconcile] Reconciled cluster state successfully
2020/02/28 00:53:39 [INFO] Setting maxUnavailable for worker nodes to: 1
2020/02/28 00:53:39 [INFO] Setting maxUnavailable for control nodes to: 1
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: Pre-pulling kubernetes images
2020/02/28 00:53:39 [INFO] Image [rancher/hyperkube:v1.16.7-rancher1] exists on host [167.172.28.157]
2020/02/28 00:53:39 [INFO] Image [rancher/hyperkube:v1.16.7-rancher1] exists on host [138.197.2.221]
2020/02/28 00:53:39 [INFO] Image [rancher/hyperkube:v1.16.7-rancher1] exists on host [167.172.26.9]
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: Kubernetes images pulled successfully
2020/02/28 00:53:39 [INFO] cluster [c-p8w2x] provisioning: [etcd] Building up etcd plane..
2020/02/28 00:53:39 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.26.9]
2020/02/28 00:53:39 [INFO] Starting container [etcd-fix-perm] on host [167.172.26.9], try #1



2020/02/28 00:53:40 [INFO] cluster [c-p8w2x] provisioning: Successfully started [etcd-fix-perm] container on host [167.172.26.9]
2020/02/28 00:53:40 [INFO] cluster [c-p8w2x] provisioning: Waiting for [etcd-fix-perm] container to exit on host [167.172.26.9]
2020/02/28 00:53:40 [INFO] cluster [c-p8w2x] provisioning: Waiting for [etcd-fix-perm] container to exit on host [167.172.26.9]
2020/02/28 00:53:40 [INFO] Removing container [etcd-fix-perm] on host [167.172.26.9], try #1



2020/02/28 00:53:40 [INFO] cluster [c-p8w2x] provisioning: [remove/etcd-fix-perm] Successfully removed container on host [167.172.26.9]
2020/02/28 00:53:41 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.26.9]
2020/02/28 00:53:41 [INFO] Starting container [rke-log-linker] on host [167.172.26.9], try #1



2020/02/28 00:53:41 [INFO] cluster [c-p8w2x] provisioning: [etcd] Successfully started [rke-log-linker] container on host [167.172.26.9]
2020/02/28 00:53:42 [INFO] Removing container [rke-log-linker] on host [167.172.26.9], try #1



2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [remove/rke-log-linker] Successfully removed container on host [167.172.26.9]
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [etcd] Successfully started etcd plane.. Checking etcd cluster health
2020/02/28 00:53:42 [INFO] [controlplane] Now checking status of node sowmya-ctrl-2
2020/02/28 00:53:42 [INFO] [controlplane] Now checking status of node sowmya-ctrl-3
2020/02/28 00:53:42 [INFO] Attempting upgrade of controlplane components on following hosts in NotReady status: 
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [controlplane] Building up Controller Plane..
2020/02/28 00:53:42 [INFO] Resetting maxUnavailable to 0 since 1 host(s) are found to be inactive/unavailable prior to upgrade
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [controlplane] Successfully started Controller Plane..
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [controlplane] Processing controlplane hosts for upgrade one at a time
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [controlplane] Successfully upgraded Controller Plane..
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [authz] Creating rke-job-deployer ServiceAccount
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [authz] rke-job-deployer ServiceAccount created successfully
2020/02/28 00:53:42 [INFO] cluster [c-p8w2x] provisioning: [authz] Creating system:node ClusterRoleBinding
2020/02/28 00:53:43 [INFO] cluster [c-p8w2x] provisioning: [authz] system:node ClusterRoleBinding created successfully
2020/02/28 00:53:43 [INFO] cluster [c-p8w2x] provisioning: [authz] Creating kube-apiserver proxy ClusterRole and ClusterRoleBinding
2020/02/28 00:53:43 [INFO] cluster [c-p8w2x] provisioning: [authz] kube-apiserver proxy ClusterRole and ClusterRoleBinding created successfully
2020/02/28 00:53:43 [INFO] cluster [c-p8w2x] provisioning: Successfully Deployed state file at [management-state/rke/rke-250920167/cluster.rkestate]
2020/02/28 00:53:43 [INFO] cluster [c-p8w2x] provisioning: [state] Saving full cluster state to Kubernetes
2020/02/28 00:53:43 [INFO] [worker] Now checking status of node sowmya-etcd-1
2020/02/28 00:53:43 [INFO] cluster [c-p8w2x] provisioning: [state] Successfully Saved full cluster state to Kubernetes ConfigMap: cluster-state
2020/02/28 00:53:44 [INFO] Attempting upgrade of worker components on following hosts in NotReady status: 
2020/02/28 00:53:44 [INFO] Resetting maxUnavailable to 0 since 1 host(s) are found to be inactive/unavailable prior to upgrade
2020/02/28 00:53:44 [INFO] cluster [c-p8w2x] provisioning: [worker] Building up Worker Plane..
2020/02/28 00:53:44 [INFO] cluster [c-p8w2x] provisioning: [worker] Successfully started Worker Plane..
2020/02/28 00:53:44 [INFO] cluster [c-p8w2x] provisioning: [worker] Upgrading Worker Plane..
2020/02/28 00:53:44 [INFO] cluster [c-p8w2x] provisioning: First checking and processing worker components for upgrades on nodes with etcd role one at a time
2020/02/28 00:53:44 [INFO] [workerplane] Processing host sowmya-etcd-1
2020/02/28 00:53:44 [INFO] [worker] Now checking status of node sowmya-etcd-1
2020/02/28 00:53:44 [INFO] [worker] Getting list of nodes for upgrade
2020/02/28 00:53:44 [INFO] [workerplane] Upgrade not required for worker components of host sowmya-etcd-1
2020/02/28 00:53:44 [INFO] cluster [c-p8w2x] provisioning: [worker] Successfully upgraded Worker Plane..
2020/02/28 00:53:45 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.28.157]
2020/02/28 00:53:45 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [138.197.2.221]
2020/02/28 00:53:45 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [167.172.26.9]
2020/02/28 00:53:45 [INFO] Starting container [rke-log-cleaner] on host [167.172.28.157], try #1



2020/02/28 00:53:45 [INFO] Starting container [rke-log-cleaner] on host [138.197.2.221], try #1



2020/02/28 00:53:45 [INFO] Starting container [rke-log-cleaner] on host [167.172.26.9], try #1



2020/02/28 00:53:45 [INFO] cluster [c-p8w2x] provisioning: [cleanup] Successfully started [rke-log-cleaner] container on host [167.172.28.157]
2020/02/28 00:53:45 [INFO] cluster [c-p8w2x] provisioning: [cleanup] Successfully started [rke-log-cleaner] container on host [167.172.26.9]
2020/02/28 00:53:45 [INFO] cluster [c-p8w2x] provisioning: [cleanup] Successfully started [rke-log-cleaner] container on host [138.197.2.221]
2020/02/28 00:53:45 [INFO] Removing container [rke-log-cleaner] on host [167.172.28.157], try #1



2020/02/28 00:53:45 [INFO] Removing container [rke-log-cleaner] on host [167.172.26.9], try #1



2020/02/28 00:53:45 [INFO] Removing container [rke-log-cleaner] on host [138.197.2.221], try rancher/rancher#1
2020/02/28 00:53:45 [INFO] cluster [c-p8w2x] provisioning: [remove/rke-log-cleaner] Successfully removed container on host [167.172.28.157]
2020/02/28 00:53:45 [INFO] cluster [c-p8w2x] provisioning: [remove/rke-log-cleaner] Successfully removed container on host [167.172.26.9]
2020/02/28 00:53:45 [INFO] cluster [c-p8w2x] provisioning: [remove/rke-log-cleaner] Successfully removed container on host [138.197.2.221]
2020/02/28 00:53:45 [INFO] cluster [c-p8w2x] provisioning: [sync] Syncing nodes Labels and Taints
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [sync] Successfully synced nodes Labels and Taints
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [network] Setting up network plugin: canal
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [addons] Saving ConfigMap for addon rke-network-plugin to Kubernetes
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [addons] Successfully saved ConfigMap for addon rke-network-plugin to Kubernetes
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [addons] Executing deploy job rke-network-plugin
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [addons] Setting up coredns
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [addons] Saving ConfigMap for addon rke-coredns-addon to Kubernetes
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [addons] Successfully saved ConfigMap for addon rke-coredns-addon to Kubernetes
2020/02/28 00:53:46 [INFO] cluster [c-p8w2x] provisioning: [addons] Executing deploy job rke-coredns-addon
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] CoreDNS deployed successfully..
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [dns] DNS provider coredns deployed successfully
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Setting up Metrics Server
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Saving ConfigMap for addon rke-metrics-addon to Kubernetes
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Successfully saved ConfigMap for addon rke-metrics-addon to Kubernetes
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Executing deploy job rke-metrics-addon
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Metrics Server deployed successfully
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [ingress] Setting up nginx ingress controller
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Saving ConfigMap for addon rke-ingress-controller to Kubernetes
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Successfully saved ConfigMap for addon rke-ingress-controller to Kubernetes
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Executing deploy job rke-ingress-controller
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [ingress] ingress controller nginx deployed successfully
2020/02/28 00:53:47 [INFO] kontainerdriver rancherkubernetesengine stopped
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] Setting up user addons
2020/02/28 00:53:47 [INFO] cluster [c-p8w2x] provisioning: [addons] no user addons defined
2020/02/28 00:53:47 [ERROR] ClusterController c-p8w2x [cluster-provisioner-controller] failed with : Provisioning incomplete, host(s) [167.172.249.119] skipped because they could not be contacted

```
**Expected Result:**
Add ons should get executed after the control plane has finished upgrading.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): master -head commit id: `32f6f659c`
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): rke DO
- Kubernetes version (use `kubectl version`):

```
1.15 to 1.16
```

**Comment by mrajashree** [comment]
It's because of 
```
2020/02/28 00:53:44 [INFO] Resetting maxUnavailable to 0 since 1 host(s) are found to be inactive/unavailable prior to upgrade
```

**Comment by sowmyav27** [comment]
**Verified on latest master-head commit id: `fa96c0c3d`**

- Deploy a cluster - 1 etcd, 3 control plane and 2 worker nodes.
- Power down one control plane node.
- Upgrade k8s version in the cluster from 1.15 to 1.16
- The cluster goes to Upgrading state.
- The cluster is stuck in Updating state with error: `[controlPlane] Failed to upgrade Control Plane: cannot proceed with upgrade of controlplane since 1 host(s) are found to be inactive prior to upgrade; [controlPlane] Failed to upgrade Control Plane: cannot proceed with upgrade of controlplane since 1 host(s) are found to be inactive prior to upgrade`
- Add-ons do NOT executed
- Logs:

```
2020/03/02 17:13:12 [INFO] cluster [c-52kbs] provisioning: [etcd] Building up etcd plane..
2020/03/02 17:13:12 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [<etcd-ip-addr>]
2020/03/02 17:13:12 [INFO] Starting container [etcd-fix-perm] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:13 [INFO] cluster [c-52kbs] provisioning: Successfully started [etcd-fix-perm] container on host [<etcd-ip-addr>]
2020/03/02 17:13:13 [INFO] cluster [c-52kbs] provisioning: Waiting for [etcd-fix-perm] container to exit on host [<etcd-ip-addr>]
2020/03/02 17:13:13 [INFO] cluster [c-52kbs] provisioning: Waiting for [etcd-fix-perm] container to exit on host [<etcd-ip-addr>]
2020/03/02 17:13:13 [INFO] cluster [c-52kbs] provisioning: Container [etcd-fix-perm] is still running on host [<etcd-ip-addr>]
2020/03/02 17:13:14 [INFO] cluster [c-52kbs] provisioning: Waiting for [etcd-fix-perm] container to exit on host [<etcd-ip-addr>]
2020/03/02 17:13:15 [INFO] Removing container [etcd-fix-perm] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:15 [INFO] cluster [c-52kbs] provisioning: [remove/etcd-fix-perm] Successfully removed container on host [<etcd-ip-addr>]
2020/03/02 17:13:15 [INFO] Checking if container [etcd] is running on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:15 [INFO] Pulling image [rancher/coreos-etcd:v3.3.15-rancher1] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:18 [INFO] Image [rancher/coreos-etcd:v3.3.15-rancher1] exists on host [<etcd-ip-addr>]
2020/03/02 17:13:18 [INFO] Checking if container [old-etcd] is running on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:18 [INFO] Stopping container [etcd] on host [<etcd-ip-addr>] with stopTimeoutDuration [5s], try #1
2020/03/02 17:13:19 [INFO] cluster [c-52kbs] provisioning: Waiting for [etcd] container to exit on host [<etcd-ip-addr>]
2020/03/02 17:13:19 [INFO] Renaming container [etcd] to [old-etcd] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:19 [INFO] Starting container [etcd] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:20 [INFO] Removing container [old-etcd] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:20 [INFO] cluster [c-52kbs] provisioning: [etcd] Successfully updated [etcd] container on host [<etcd-ip-addr>]
2020/03/02 17:13:21 [INFO] Image [rancher/rke-tools:v0.1.52] exists on host [<etcd-ip-addr>]
2020/03/02 17:13:21 [INFO] Starting container [rke-log-linker] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:22 [INFO] cluster [c-52kbs] provisioning: [etcd] Successfully started [rke-log-linker] container on host [<etcd-ip-addr>]
2020/03/02 17:13:22 [INFO] Removing container [rke-log-linker] on host [<etcd-ip-addr>], try #1
2020/03/02 17:13:22 [INFO] cluster [c-52kbs] provisioning: [remove/rke-log-linker] Successfully removed container on host [<etcd-ip-addr>]
2020/03/02 17:13:22 [INFO] cluster [c-52kbs] provisioning: [etcd] Successfully started etcd plane.. Checking etcd cluster health
E0302 17:13:27.755101      26 watcher.go:214] watch chan error: etcdserver: mvcc: required revision has been compacted
2020/03/02 17:13:29 [INFO] [controlplane] Now checking status of node sowmya-test-ctrl-3
2020/03/02 17:13:29 [INFO] [controlplane] Now checking status of node sowmya-test-ctrl-2
2020/03/02 17:13:30 [INFO] Attempting upgrade of controlplane components on following hosts in NotReady status: 
2020/03/02 17:13:30 [INFO] cluster [c-52kbs] provisioning: [controlplane] Building up Controller Plane..
2020/03/02 17:13:30 [INFO] kontainerdriver rancherkubernetesengine stopped
2020/03/02 17:13:30 [INFO] cluster [c-52kbs] provisioning: [controlplane] Successfully started Controller Plane..
2020/03/02 17:13:30 [INFO] cluster [c-52kbs] provisioning: [controlplane] Processing controlplane hosts for upgrade one at a time
2020/03/02 17:13:30 [ERROR] ClusterController c-52kbs [cluster-provisioner-controller] failed with : [controlPlane] Failed to upgrade Control Plane: cannot proceed with upgrade of controlplane since 1 host(s) are found to be inactive prior to upgrade
```
